src/test/java/com/moneylendingapp/config/MySqlTestContainerSetup.java

```diff
@@ -0,0 +1,29 @@
+package com.moneylendingapp.config;
+
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.context.annotation.Profile;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+@Profile("test")
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
```

Comment:
You don't need this.
The test container can pick up your mysql configuration in your properties file and use it. You don't need this class.

Just make sure your:

```
spring:
  datasource:
    url: jdbc:tc:mysql:5.7.37:////money_lending_app
    username: admin
    password: admin
    maxConnections: 10
    hikari:
      connection-test-query: SELECT 1
    driver-class-name: org.testcontainers.jdbc.ContainerDatabaseDriver
```